crates/dbsp/src/circuit/circuit_builder.rs

```diff
@@ -4935,22 +5843,357 @@ impl CircuitHandle {
                         elapsed.as_secs_f64()
                     );
                 }
-            });
-        Ok(())
+
+                Ok(())
+            })
     }
 
-    pub fn restore(&mut self, base: &StoragePath) -> Result<(), SchedulerError> {
-        self.circuit
-            .map_nodes_recursive_mut(&mut |node: &mut dyn Node| {
-                node.restore(base).expect("restored");
-            });
+    /// Restores the circuit from a checkpoint.
+    ///
+    /// Restore the circuit from a checkpoint and prepare it to backfill new and
+    /// mofified parts of the circuit if necessary.
```

Comment:
modified
my editor does spell-checking in comments too.
maybe you can set up yours as well?